x/stakeibc/keeper/update_redemption_rates_test.go

```diff
@@ -0,0 +1,272 @@
+	STAKED_BAL := genRandUintBelowMax(MAX)
+	UNDELEGATED_BAL := genRandUintBelowMax(MAX)
+	JUST_DEPOSITED_BAL := genRandUintBelowMax(MAX)
+
+	ST_SUPPLY := genRandUintBelowMax(MAX)
+
+	// s.Require().ElementsMatch([]int{0, 0, 0, 0}, []int{int(STAKED_BAL), int(UNDELEGATED_BAL), int(JUST_DEPOSITED_BAL), int(ST_SUPPLY)}) //
+	REDEMPTION_RATE_T0 := sdk.NewDec(1)
+	tc := s.SetupUpdateRedemptionRates(STAKED_BAL, UNDELEGATED_BAL, JUST_DEPOSITED_BAL, ST_SUPPLY, REDEMPTION_RATE_T0)
+
+	// sanity check on inputs (check redemptionRate at genesis is 1)
+	s.Require().Equal(tc.redemption_rate_t0, sdk.NewDec(1))
+
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx)
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
+
+	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.host_zone.ChainId)
+	s.Require().True(found)
+	rrNew := hz.RedemptionRate
+
+	numerator := int64(STAKED_BAL) + int64(UNDELEGATED_BAL) + int64(JUST_DEPOSITED_BAL)
+	denominator := int64(ST_SUPPLY)
+	expectedNewRate := sdk.NewDec(numerator).Quo(sdk.NewDec(denominator))
+	s.Require().Equal(rrNew, expectedNewRate)
+}
+
+func (s *KeeperTestSuite) TestUpdateRedemptionRatesRandomized_MultipleRuns() {
+	for i := 0; i < 100; i++ {
+		s.TestUpdateRedemptionRatesRandomized()
+		// reset the testing app between runs
+		s.Setup()
+	}
+}
+
+func (s *KeeperTestSuite) TestUpdateRedemptionRateZeroStAssets() {
+
+	STAKED_BAL := uint64(5)
+	UNDELEGATED_BAL := uint64(3)
+	JUST_DEPOSITED_BAL := uint64(3)
+	ST_SUPPLY := uint64(0)
+
+	REDEMPTION_RATE_T0 := sdk.NewDec(1)
+	tc := s.SetupUpdateRedemptionRates(STAKED_BAL, UNDELEGATED_BAL, JUST_DEPOSITED_BAL, ST_SUPPLY, REDEMPTION_RATE_T0)
+
+	// sanity check on inputs (check redemptionRate at genesis is 1)
+	s.Require().Equal(tc.redemption_rate_t0, sdk.NewDec(1))
+
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx)
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
+
+	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.host_zone.ChainId)
+	s.Require().True(found)
+	rrNew := hz.RedemptionRate
+
+	// RR should be unchanged
+	s.Require().Equal(rrNew, sdk.NewDec(1))
+}
+
+func (s *KeeperTestSuite) TestUpdateRedemptionRateZeroNativeAssets() {
+
+	STAKED_BAL := uint64(0)
+	UNDELEGATED_BAL := uint64(0)
+	JUST_DEPOSITED_BAL := uint64(0)
+	ST_SUPPLY := uint64(10)
+
+	REDEMPTION_RATE_T0 := sdk.NewDec(1)
+	tc := s.SetupUpdateRedemptionRates(STAKED_BAL, UNDELEGATED_BAL, JUST_DEPOSITED_BAL, ST_SUPPLY, REDEMPTION_RATE_T0)
+
+	// sanity check on inputs (check redemptionRate at genesis is 1)
+	s.Require().Equal(tc.redemption_rate_t0, sdk.NewDec(1))
+
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx)
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
+
+	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.host_zone.ChainId)
+	s.Require().True(found)
+	rrNew := hz.RedemptionRate
+
+	// RR should be 0
+	s.Require().Equal(rrNew, sdk.NewDec(0))
+}
+
+func (s *KeeperTestSuite) TestUpdateRedemptionRate_NoModuleAccountRecords() {
```

Comment:
nit: some function names use the underscore and some don't - I think either is fine, but just keep it consistent

Reply:
done